x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/script/ScriptTemplate.java

```diff
@@ -26,15 +30,25 @@
     private final Params params;
     // used for sorting based on scripts
     private final DataType outputType;
+    private final List<ScriptFunction> functions;
 
     public ScriptTemplate(String template) {
         this(template, Params.EMPTY, DataType.KEYWORD);
     }
 
     public ScriptTemplate(String template, Params params, DataType outputType) {
+        this(template, params, emptyList(), outputType);
+    }
+
+    public ScriptTemplate(String template, Params params, ScriptFunction function, DataType outputType) {
```

Comment:
I'd prefer not to have so many ctors. Could you use `singletonList(function)` instead of this? I'd prefer not to have all the old ctors too but that sounds like a good thing to do in a followup.